File: backend/app/schemas/staff_attendance.py
"""Pydantic request/response schemas for staff/trainer attendance."""

from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class StaffCheckInRequest(BaseModel):
    """Payload to check a staff/trainer user in for the current day.

    ``staff_id`` is optional — omit it to check the authenticated caller
    themselves in, or provide it (e.g. from a front-desk search) to check
    in a different staff/trainer user.
    """

    staff_id: int | None = Field(
        default=None, gt=0, description="Staff user to check in; defaults to the caller"
    )


class StaffAttendanceResponse(BaseModel):
    """Public representation of a single staff attendance record."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    staff_id: int
    staff_name: str
    check_in_time: datetime
    check_out_time: datetime | None
    date: date
    created_at: datetime

File: frontend/src/types/staffAttendance.ts
/**
 * Types for the Staff Attendance module, mirroring
 * backend/app/schemas/staff_attendance.py and staff.py.
 */

export type StaffRole = 'owner' | 'admin' | 'staff' | 'trainer';

export interface StaffListItem {
  id: number;
  full_name: string;
  role: StaffRole;
  is_active: boolean;
}

export interface StaffAttendanceRecord {
  id: number;
  staff_id: number;
  staff_name: string;
  check_in_time: string;
  check_out_time: string | null;
  date: string;
  created_at: string;
}

/** Payload for `POST /api/v1/staff-attendance/check-in`. */
export interface StaffCheckInPayload {
  staff_id?: number;
}
